Add breathe example

dwillmore opened the pull request.

dwillmore commented:
All the checks passed, merge?

cnlohr commented on examples/breathe/breathe.c:
```diff
@@ -0,0 +1,34 @@
+	int32_t io = 1;
+
+	SystemInit();
+	RCC->APB2PCENR |= RCC_APB2Periph_GPIOD;
```
Please add comment showing what you are causing to be hidden here.  This is a good trick to learn but not obvious that you are replacing funConfig lines.

dwillmore replied:
I added a large block of comments to the head of the file and converted to the fun version of the config and GPIO functions.

cnlohr commented on examples/breathe/breathe.c:
```diff
@@ -0,0 +1,34 @@
+#include "ch32fun.h"
```
Please add a comment here explaining what this is doing/why.

dwillmore replied:
Added.

cnlohr commented on examples/breathe/breathe.c:
```diff
@@ -0,0 +1,34 @@
+		if((volume == 1) || (volume == 65535))
+			io = -io;
+		else
+		  if(io == 1)
```
Ok, this needs a fair bit of explanation...

Also, because you are doubly nesting if's can you use {} 's?

dwillmore commented:
Added a large header comment to explain the purpose and operation of the code

Converted to fun version of GPIO functions for port initialization and manipulation.

Interspersed comments to explain code flow and function

Added braces to a nested if statement for clarity.
